### PR TITLE
Job to publish a nightly build

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
It would be useful to have a public nightly build.

# Changes
* Added a script to build a package as a tar file.
* Added a GitHub job to build and publish.

# TODO
* Add a GitHub org secret
* When there is a secret in place, schedule this to run nightly.

# Tests
* The tarfile build works.
* The github job is untested.  I don't have permissions to add an org secret.  We can run the publishing step manually for now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dfinity/nns",
-  "version": "0.2.2",
+  "version": "nightly-2022-02-28",
```
Comment (peterpeterparker) [inline package.json:3]: Should the version be commited? I would say we keep the official minor-major version in the repo spontaneously but just my two cents.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  formatting:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build nightly
+        run: ./scripts/build-nightly
+      - name: Publish
+        run: npm publish
```
Comment (peterpeterparker) [inline .github/workflows/nightly.yml:16]: Do we want to publish version under a `nightly` flag?

```
npm publish --tag nightly
```

https://docs.npmjs.com/adding-dist-tags-to-packages

**Review by peterpeterparker — CHANGES_REQUESTED** [review]
Super cool!!!!!!!

Two questions.

File: package.json (hunk 1 above)
Comment (bitdivine) [reply package.json:3]: No, this is an accident.  Thanks!

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (bitdivine) [reply .github/workflows/nightly.yml]: Good idea.  Done.  What was the second question?

**Comment by lmuntanerDfinity** [comment]
🚀

**Review by lmuntanerDfinity — APPROVED** [review]

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx!

**Pull request merged**